app/models/ems_event.rb

```diff
@@ -84,6 +84,10 @@ def self.add_kubernetes(ems_id, event)
     add(ems_id, ManageIQ::Providers::Kubernetes::ContainerManager::EventParser.event_to_hash(event, ems_id))
   end
 
+  def self.add_middleware(ems_id, event)
```

Comment:
yes, I moved it to the EventCatcher
https://github.com/ManageIQ/manageiq/pull/7372/files
@jshaughn could you also follow this pattern?